app/views/comments/_form.html.erb

```diff
@@ -0,0 +1,17 @@
+<%= form_with model: [ @article, @article.comments.build ] do |form| %>
```

Comment:
@JuanMaGiraldo how the `form_with model:` works?

Reply:
form_with helps us to create forms efficiently, eliminating the complexity of creating them. There are different ways to use them, in the case of using the model it takes care of supplying the url automatically. If we were editing an article, rails would transform this:
`<% = form_with model: @article do | form | %> `to this:
`<form action ="/articles/id" method = "post " accept-charset = "UTF-8">`

In this case, a single object is not sent to the model, but an array of objects is sent.  The first object which is the article, would be composing the first part of the path for the form which would be `/articles / id`, then the second object is a new comment created with the method .build in the array of comments in the article and it would finish completing the route with `/comments`, leaving the following form: `/articles/id/comments`